lemur/auth/views.py

```diff
@@ -278,6 +283,43 @@ def update_user(user, profile, roles):
     return user
 
 
+def build_hmac():
+    key = current_app.config.get('OAUTH_STATE_TOKEN_SECRET', OAUTH_STATE_TOKEN_SECRET_FALLBACK)
+    return hmac.HMAC(key, hashes.SHA256(), backend=default_backend())
+
+
+def generate_state_token():
+    t = int(time.time())
+    ts = hex(t)[2:].encode('ascii')
+    h = build_hmac()
+    h.update(ts)
+    digest = base64.b64encode(h.finalize())
+    state = ts + b':' + digest
+    return state.decode('utf-8')
+
+
+def verify_state_token(token):
+    stale_seconds = current_app.config.get('OAUTH_STATE_TOKEN_STALE_TOLERANCE_SECONDS', 15)
+    try:
+        state = token.encode('utf-8')
+        ts, digest = state.split(b':')
```

Comment:
to render this more robust, maybe it would be helpful to catch this error, by first checking the correctness of the format?

```
ValueError: not enough values to unpack (expected 2, got 1)
```

Reply:
Thanks, I split up the token parsing logic from the verification work - commit forthcoming!